agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/AgentImplementation.java

```diff
@@ -167,7 +167,7 @@ public static String getAgentJarLocation() throws UnsupportedEncodingException {
         }
 
         String stringPath = AgentImplementation.class.getProtectionDomain().getCodeSource().getLocation().getPath();
-        return URLDecoder.decode(stringPath, "UTF-8");
+        return new File(URLDecoder.decode(stringPath, "UTF-8")).getParent();
```

Comment:
Does this change work for JDK 8 as well? Or it will break JDK 8?

Reply:
JDK 8 will use the URLClassLoader a few lines up.

Not sure what this fall through case was for before, any idea?

Reply:
Ah, I see. Yes, JDK8 will use the if-block above with URL class loader.

Reply:
I'm not really sure either. I think the CodeSource method would work in either case. This looks fine for now.